File: ProgrammingFundamentals/Java/02.ConditionalStatementsAndLoops/Exercises/src/P07_CakeIngredients.java
import java.util.Scanner;
public class P07_CakeIngredients {
    public static void main(String[] args) {
        Scanner console = new Scanner(System.in);
        for(int i = 0; i <= 20; i++) {
            String ingredient = console.nextLine();
            if (ingredient.equals("Bake!")) {
                System.out.format("Preparing cake with %d ingredients.", i);
                break;
            }
            System.out.format("Adding ingredient %s." + System.lineSeparator(), ingredient);
        }
    }
}

File: ProgrammingFundamentals/C#/02.ConditionalStatementsAndLoops/Exercises/P07_CakeIngredients/P07_CakeIngredients.cs
using System;

namespace P07_CakeIngredients
{
    class P07_CakeIngredients
    {
        static void Main(string[] args)
        {
            for (int i = 0; i <= 20; i++)
            {
                string ingredient = Console.ReadLine();
                if (ingredient == "Bake!")
                {
                    Console.WriteLine($"Preparing cake with {i} ingredients.");
                    break;
                }
                Console.WriteLine($"Adding ingredient {ingredient}.");
            }
        }
    }
}
